src/Validation/EmptyConfigurableRectorChecker.php

```diff
@@ -0,0 +1,125 @@
+        }
+
+        $this->reportWarningMessage($emptyConfigurableRectors);
+        $this->reportEmptyConfigurableMessage($emptyConfigurableRectors);
+    }
+
+    /**
+     * @param RectorInterface[] $rectors
+     * @return RectorInterface[]
+     */
+    private function resolveEmptyConfigurable(array $rectors): array
+    {
+        $emptyConfigurableRectors = [];
+        foreach ($rectors as $rector) {
+            if ($this->shouldSkip($rector)) {
+                continue;
+            }
+
+            $ruleDefinition = $rector->getRuleDefinition();
+
+            /** @var CodeSampleInterface[] $codeSamples */
+            $codeSamples = $ruleDefinition->getCodeSamples();
+            foreach ($codeSamples as $codeSample) {
+                $configuration = $codeSample->getConfiguration();
+                if (! is_array($configuration)) {
+                    continue;
+                }
+
+                foreach ($configuration as $key => $config) {
+                    if (! is_array($config)) {
+                        continue;
+                    }
+
+                    try {
+                        $key = $this->propertyNaming->underscoreToName($key);
+                        $value = $this->privatesAccessor->getPrivateProperty($rector, $key);
+
+                        if (is_array($value) && $value === []) {
+                            $emptyConfigurableRectors[] = $rector;
+                            continue 3;
+                        }
+                    } catch (Throwable) {
+                        continue;
+                    }
+                }
+            }
+        }
+
+        return $emptyConfigurableRectors;
+    }
+
+    private function shouldSkip(RectorInterface $rector): bool
+    {
+        if (! $rector instanceof ConfigurableRectorInterface) {
+            return true;
+        }
+
+        // it seems always loaded
+        return $rector instanceof RenameClassNonPhpRector;
+    }
+
+    /**
+     * @param RectorInterface[] $emptyConfigurableRectors
+     */
+    private function reportWarningMessage(array $emptyConfigurableRectors): void
+    {
+        $warningMessage = sprintf(
+            'Your project requires %d configurable rector rules that needs to be configured, use -vvv for detailed info.',
+            count($emptyConfigurableRectors)
+        );
+
+        $this->symfonyStyle->warning($warningMessage);
+    }
+
+    /**
+     * @param RectorInterface[] $emptyConfigurableRectors
+     */
+    private function reportEmptyConfigurableMessage(array $emptyConfigurableRectors): void
+    {
+        if (! $this->symfonyStyle->isVerbose()) {
+            return;
+        }
+
+        foreach ($emptyConfigurableRectors as $emptyConfigurableRector) {
+            $shortRectorClass = Strings::after($emptyConfigurableRector::class, '\\', -1);
+
+            $rectorMessage = ' * ' . $shortRectorClass;
+            $this->symfonyStyle->writeln($rectorMessage);
+        }
+    }
+}
```

Comment:
Thanks :+1: 

I think this might break in the future - the `resolveEmptyConfigurable()` method should be tested. Could you add one test case with 2 fixtures:

* one that reports missing configurable rule
* and other passing correctly?